File: 2427-first-letter-to-appear-twice/first-letter-to-appear-twice.java
class Solution {
    public char repeatedCharacter(String s) {
               // 26 size ka integer array banaya, shuru me sabhi dabbe me 0 hoga
        int[] letterCounts = new int[26]; // camelCase
        
        for (int i = 0; i < s.length(); i++) {
            char currentCharacter = s.charAt(i); // camelCase
            int alphabetIndex = currentCharacter - 'a'; // camelCase
            
            // Har baar character aane par uski ginti badhao
            letterCounts[alphabetIndex]++;
            
            // Agar ginti 2 ho gayi, matlab yeh pehla letter hai jo doosri baar dikha
            if (letterCounts[alphabetIndex] == 2) {
                return currentCharacter;
            }
        }
        
        return ' ';
    }
}

File: 2427-first-letter-to-appear-twice/first-letter-to-appear-twice.cpp
class Solution {
public:
    char repeatedCharacter(string s) {
         // 26 size ka vector jisme shuru me sabhi values 0 hain
        vector<int> letter_counts(26, 0); // snake_case
        
        for (int i = 0; i < s.length(); i++) {
            char current_character = s[i]; // snake_case
            int alphabet_index = current_character - 'a'; // snake_case
            
            // Ginti ko 1 badhaya
            letter_counts[alphabet_index]++;
            
            // Check kiya ki kya yeh doosri baar aaya hai
            if (letter_counts[alphabet_index] == 2) {
                return current_character;
            }
        }
        
        return ' ';
    }
};